src/status_im/subs.cljs

```diff
@@ -1868,3 +1831,68 @@
                             (or (string/blank? screen-snt-amount)
                                 (#{"0" "0.0" "0.00"} screen-snt-amount)
                                 (string/ends-with? screen-snt-amount ".")))))))))
+
+;;SIGNING =============================================================================================================
+
+(re-frame/reg-sub
+ :signing/fee
+ :<- [:signing/tx]
+ (fn [{:keys [gas gasPrice]}]
+   (signing.gas/calculate-max-fee gas gasPrice)))
+
+(re-frame/reg-sub
+ :signing/phrase
+ :<- [:account/account]
+ (fn [{:keys [signing-phrase]}]
+   signing-phrase))
+
+(defn- too-precise-amount?
+  "Checks if number has any extra digit beyond the allowed number of decimals.
+  It does so by checking the number against its rounded value."
+  [amount decimals]
+  (let [bn (money/bignumber amount)]
+    (not (.eq bn (.round bn decimals)))))
+
+(defn get-amount-error [amount decimals]
+  (when (and (not (empty? amount)) decimals)
+    (let [normalized-amount (money/normalize amount)
+          value             (money/bignumber normalized-amount)]
+      (cond
+        (not (money/valid? value))
+        {:amount-error (i18n/label :t/validation-amount-invalid-number)}
+
+        (too-precise-amount? normalized-amount decimals)
+        {:amount-error (i18n/label :t/validation-amount-is-too-precise {:decimals decimals})}
+
+        :else nil))))
+
+(defn get-sufficient-funds-error
+  [balance symbol amount]
+  (when-not (money/sufficient-funds? amount (get balance symbol))
+    {:amount-error (i18n/label :t/wallet-insufficient-funds)}))
+
+(defn get-sufficient-gas-error
+  [balance symbol amount gas gasPrice]
+  (if (and gas gasPrice)
+    (let [fee               (.times gas gasPrice)
+          available-ether   (money/bignumber (get balance :ETH 0))
+          available-for-gas (if (= :ETH symbol)
+                              (.minus available-ether (money/bignumber amount))
+                              available-ether)]
+      (when-not (money/sufficient-funds? fee (money/bignumber available-for-gas))
+        {:gas-error (i18n/label :t/wallet-insufficient-gas)}))
+    {:gas-error (i18n/label :t/invalid-number)}))
+
+(re-frame/reg-sub
+ :signing/amount-errors
+ :<- [:signing/tx]
+ :<- [:balance]
+ (fn [[{:keys [amount token gas gasPrice]} balance]]
+   (if (and amount token)
+     (let [amount-bn (money/formatted->internal (money/bignumber amount) (:symbol token) (:decimals token))
+           amount-error (or (get-amount-error amount (:decimals token))
+                            (get-sufficient-funds-error balance (:symbol token) amount-bn))]
+       (if amount-error
+         amount-error
```

Comment:
could be shorter with `or`

`(or amount-error (get-sufficient-gas-error ...))`

Reply:
based on @rasom comments during latest dev call, don't you think these are good candidate for local component state?   while the functions themselves should stay separated from the UI and tested, does the whole app need to know about this state?